static/js/controllers/reports.js

```diff
@@ -250,17 +335,18 @@ var _ = require('underscore'),
             _query();
           }
         }
-
       };
 
       $scope.$on('ClearSelected', function() {
-        $scope.selected = null;
+        $scope.selected = [];
+        $('#reports-list input[type="checkbox"]')
+          .prop('checked', false);
         liveList.clearSelected();
```

Comment:
Because the livelist functions called and the selectors used to check the checkboxes are almost always different for the different invocations you'd end up with something like...
```
function modifySelection(liveListFunction, checkboxSelector, checked) {
  liveListFunction();
  $(checkboxSelector).prop('checked', checked);
}
```

This doesn't reduce complexity at all because all the interesting code is still in the calling function, and it just adds yet another layer of abstraction. The one benefit is it'll remind people to do both.